stdlib/string.ml

```diff
@@ -228,8 +230,9 @@ let hash x = seeded_hash 0 x
 (* duplicated in bytes.ml *)
 let split_on_char sep s =
   let r = ref [] in
-  let j = ref (length s) in
-  for i = length s - 1 downto 0 do
+  let len = length s in
+  let j = ref len in
+  for i = len - 1 downto 0 do
```

Comment:
There is still a diff here, and in particular bytes.ml and string.ml are not identical anymore. Can you revert the change? (`length` is computed twice, but this gets eliminated by CSE anyway, so it's no big deal, and slightly less work than also rephrasing bytes.ml)